.github/workflows/cuda.yml

```diff
@@ -18,16 +18,12 @@ jobs:
     env:
       CXXFLAGS: "-Werror"
       CMAKE_GENERATOR: Ninja
-      # setuptools/mp4py work-around, see
-      #   https://github.com/mpi4py/mpi4py/pull/159
-      #   https://github.com/mpi4py/mpi4py/issues/157#issuecomment-1001022274
-      SETUPTOOLS_USE_DISTUTILS: stdlib
     steps:
     - uses: actions/checkout@v3
     - uses: actions/setup-python@v4
       name: Install Python
       with:
-        python-version: '3.10'
+        python-version: '3.x'
```

Comment:
What does this do?

Reply:
Unconstrain the upper Python version that gets installed. I constrained it in https://github.com/ECP-WarpX/WarpX/commit/ed13486ac0495b049958989022a9c76e8ef8f042 to unbreak CI by avoiding Python 3.12 for a few days.

I want it unconstrained again, so I see if there are breaking changes once new Python versions are released.

Ref.: https://github.com/actions/setup-python#setup-python